src/views/CreateTransferOrder.vue

```diff
@@ -352,23 +373,103 @@ async function findProductFromIdentifier(payload: any) {
   }
 }
 
-async function addProductToCount() {
-  if (!searchedProduct.value.productId ||!queryString.value) return;
-  if (isProductAvailableInOrder()) return;
+function toggleScan() {
+  isScanningEnabled.value = !isScanningEnabled.value;
+  isScanningEnabled.value ? addProductInput.value.$el.setFocus() : addProductInput.value.$el.blur();
+}
+
+async function addProductToOrder(scannedId?: any, product?: any) {
+  if(!isScanningEnabled.value) {
+    if(!searchedProduct.value.productId ||!queryString.value) return;
+    if(isProductAvailableInOrder()) return;
+  }
 
   let newProduct = { 
-    productId: searchedProduct.value.productId,
-    sku: searchedProduct.value.sku,
+    productId: product ? product.productId :searchedProduct.value.productId,
+    sku: product ? product.sku : searchedProduct.value.sku,
     quantity: 0,
-    isChecked: false
+    isChecked: false,
+    isMatching: false,
+    noMatchFound: false
   } as any;
 
   const stock = await fetchStock(newProduct.productId);
   if(stock?.quantityOnHandTotal || stock?.quantityOnHandTotal === 0) {
     newProduct = { ...newProduct, qoh: stock.quantityOnHandTotal, atp: stock.availableToPromiseTotal }
   }
+  
+  if(product) {
+    currentOrder.value.items = currentOrder.value.items.map((item: any) => {
+      if(item.scannedId === scannedId) {
+        return newProduct;
+      }
+      return item;
+    });
+  } else {
+    currentOrder.value.items.push(newProduct);
+  }
+}
+
+// Updates the scanned product by checking if it already exists in the order and adding it if not
+async function scanProduct(scannedItem: string) {
+  if(!isScanningEnabled.value) return;
```

Comment:
I think this check is not required as we only call this method when scanning is enabled, so we don't need to check the same condition again.